sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java

```diff
@@ -182,12 +182,12 @@ private String toKeyVaultSecretName(@NonNull String property) {
             if (property.matches("[a-z0-9A-Z-]+")) {
                 return property.toLowerCase(Locale.US);
             } else if (property.matches("[A-Z0-9_]+")) {
-                return property.toLowerCase(Locale.US).replaceAll("_", "-");
+                return property.toLowerCase(Locale.US).replace("_", "-");
             } else {
                 return property.toLowerCase(Locale.US)
-                    .replaceAll("-", "") // my-project -> myproject
-                    .replaceAll("_", "") // my_project -> myproject
-                    .replaceAll("\\.", "-"); // acme.myproject -> acme-myproject
+                    .replace("-", "") // my-project -> myproject
+                    .replace("_", "") // my_project -> myproject
+                    .replace("\\.", "-"); // acme.myproject -> acme-myproject
```

Comment:
This is not correct, if we use `replaceAll` then we need the `\\` because it uses regex, but if we use `replace`, we don't need the `\\` anymore. Please add test cases for this.

Reply:
good catch.